cmake/TestImpactFramework/LYTestImpactFramework.cmake

```diff
@@ -324,31 +324,64 @@ function(ly_test_impact_write_gem_target_enumeration_file GEM_TARGET_TEMPLATE_FI
      configure_file(${GEM_TARGET_TEMPLATE_FILE} ${mapping_path})
 endfunction()
 
+#! ly_extract_aliased_target_dependencies: recursively extracts the aliases of a target to retrieve the true de-aliased target.
+#
+# \arg:TARGET target to de-alias
+function(ly_extract_aliased_target_dependencies TARGET)
+    if(ARGN STREQUAL "")
+        # Entry point of recursive call, set the parent target and clear any existing aliases
+        set(PARENT_TARGET ${TARGET})
+        set_property(GLOBAL PROPERTY LY_DE_ALIASED_TARGETS_${PARENT_TARGET} "")
```

Comment:
Why is this being cleared out?
The LyTestImpactFramework.cmake shouldn't clear out list of alias targets since it is not responsible for adding those alias targets.

Pretty much the Dealiase targets property isn't solely for the TestImpactFramework.

Instead what should be done here, is to return from the `ly_extract_aliased_target_dependencies` function.

Reply:
This is a global variable only used by `ly_extract_aliased_target_dependencies` as a fudge to facilitate the storage of common state between recursive calls. I have renamed it to `LY_EXTRACT_ALIASED_TARGET_DEPENDENCIES_DE_ALIASED_TARGETS_` (soon to be `LY_EXTRACT_ALIASED_TARGET_DEPENDENCIES_DE_ALIASED_TARGETS_`) to reflect this 👍